google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudLibraryDependencyWriter.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.cloud.tools.libraries.json.CloudLibraryClientMavenCoordinates;
+import com.intellij.openapi.application.Result;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.util.xml.DomUtil;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.idea.maven.dom.MavenDomUtil;
+import org.jetbrains.idea.maven.dom.model.MavenDomDependency;
+import org.jetbrains.idea.maven.dom.model.MavenDomProjectModel;
+import org.jetbrains.idea.maven.model.MavenId;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+
+/** A helper class that writes dependencies on Cloud libraries to a given module. */
+final class CloudLibraryDependencyWriter {
+
+  /** Prevents instantiation. */
+  private CloudLibraryDependencyWriter() {}
+
+  /**
+   * Adds the given set of {@link CloudLibrary CloudLibraries} to the given {@link Module}.
+   *
+   * <p>For a module whose project manages its dependencies through Maven, the list of libraries
+   * will be added as dependencies to the {@code pom.xml}. For all other dependency management
+   * systems, the libraries will be downloaded and added directly to the module's classpath.
+   *
+   * @param libraries the set of {@link CloudLibrary CloudLibraries} to add
+   * @param module the {@link Module} to add the libraries to
+   */
+  static void addLibraries(Set<CloudLibrary> libraries, Module module) {
```

Comment:
I think the user should get some sort of confirmation that the library was written; a simple write to the event log could work.

Reply:
did you address this?

Reply:
Done.